google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineServerModel.java

```diff
@@ -124,107 +138,524 @@ public int getDefaultPort() {
+
+  public void setPythonStartupScript(String pythonStartupScript) {
+    settings.setPythonStartupScript(pythonStartupScript);
+  }
+
+  @Override
+  public String getPythonStartupArgs() {
+    return settings.getPythonStartupArgs();
+  }
+
+  public void setPythonStartupArgs(String pythonStartupArgs) {
+    settings.setPythonStartupArgs(pythonStartupArgs);
+  }
+
+  private static final String SEPARATOR = "!@#";
+
+  @Override
+  public List<String> getJvmFlags() {
+    return Arrays.asList(settings.getJvmFlags() != null
+        ? settings.getJvmFlags().split(SEPARATOR) : new String[]{});
+  }
+
+  public void addJvmFlag(String flag) {
+    settings.setJvmFlags(settings.getJvmFlags() == null
+        ? flag : settings.getJvmFlags() + SEPARATOR + flag);
+  }
+
+  public void addAllJvmFlags(Collection<String> flags) {
+    settings.setJvmFlags(settings.getJvmFlags() == null
+        ? Joiner.on(SEPARATOR).join(flags)
+        : settings.getJvmFlags() + SEPARATOR + Joiner.on(SEPARATOR).join(flags));
+  }
+
+  public void setJvmFlags(String jvmFlags) {
+    settings.setJvmFlags(jvmFlags);
+  }
+
+  @Override
+  public String getCustomEntrypoint() {
+    return settings.getCustomEntrypoint();
+  }
+
+  public void setCustomEntrypoint(String customEntrypoint) {
+    settings.setCustomEntrypoint(customEntrypoint);
+  }
+
+  @Override
+  public String getRuntime() {
+    return settings.getRuntime();
+  }
+
+  public void setRuntime(String runtime) {
+    settings.setRuntime(runtime);
+  }
+
+  @Override
+  public Boolean getAllowSkippedFiles() {
+    return settings.isAllowSkippedFiles();
+  }
+
+  public void setAllowSkippedFiles(Boolean allowSkippedFiles) {
+    settings.setAllowSkippedFiles(allowSkippedFiles);
+  }
+
+  @Override
+  public Integer getApiPort() {
+    return settings.getApiPort();
+  }
+
+  public void setApiPort(Integer apiPort) {
+    settings.setApiPort(apiPort);
+  }
+
+  @Override
+  public Boolean getAutomaticRestart() {
+    return settings.isAutomaticRestart();
+  }
+
+  public void setAutomaticRestart(Boolean automaticRestart) {
+    settings.setAutomaticRestart(automaticRestart);
+  }
+
+  @Override
+  public String getDevAppserverLogLevel() {
+    return settings.getDevAppserverLogLevel();
+  }
+
+  public void setDevAppserverLogLevel(String devAppserverLogLevel) {
+    settings.setDevAppserverLogLevel(devAppserverLogLevel);
+  }
+
+  @Override
+  public Boolean getSkipSdkUpdateCheck() {
+    return settings.isSkipSdkUpdateCheck();
+  }
+
+  public void setSkipSdkUpdateCheck(Boolean skipSdkUpdateCheck) {
+    settings.setSkipSdkUpdateCheck(skipSdkUpdateCheck);
+  }
+
+  @Override
+  public String getDefaultGcsBucketName() {
+    return settings.getDefaultGcsBucketName();
+  }
+
+  public void setDefaultGcsBucketName(String defaultGcsBucketName) {
+    settings.setDefaultGcsBucketName(defaultGcsBucketName);
+  }
+
+  /**
+   * This class is used to serialize run/debug config settings. It only supports basic types (e.g.,
```

Comment:
I had a hard time choosing this word because I'm not too sure what can and can't be serialized. I had issues serializing lists and I read somewhere that custom objects couldn't be serialized by XmlSerializer. Didn't want to use primitive type because String isn't primitive.